swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/data/BillableUsageRemittanceRepository.java

```diff
@@ -140,6 +142,69 @@ public int resetBillableUsageRemittance(
     return update(query, parameters);
   }
 
+  public List<BillableUsageRemittanceEntity> updateStaleInProgress(long days, OffsetDateTime now) {
+    return updateStatusForStaleRemittances(
+        days,
+        RemittanceStatus.IN_PROGRESS,
+        RemittanceStatus.FAILED,
+        RemittanceErrorCode.SENDING_TO_AGGREGATE_TOPIC,
+        now);
+  }
+
+  public List<BillableUsageRemittanceEntity> updateStaleSent(long days, OffsetDateTime now) {
+    return updateStatusForStaleRemittances(
+        days, RemittanceStatus.SENT, RemittanceStatus.UNKNOWN, null, now);
+  }
+
+  @Transactional
+  public List<BillableUsageRemittanceEntity> updateStatusForStaleRemittances(
+      long days,
+      RemittanceStatus oldStatus,
+      RemittanceStatus newStatus,
+      RemittanceErrorCode errorCode,
+      OffsetDateTime now) {
+    OffsetDateTime cutoffDate = now.minusDays(days);
+
+    String query;
+    Map<String, Object> parameters = new HashMap<>();
+
+    query =
+        "update BillableUsageRemittanceEntity bu "
+            + "set bu.status = :newStatus, "
+            + "bu.errorCode = :errorCode, "
+            + "bu.updatedAt = :updatedAt "
+            + "where bu.status = :oldStatus "
+            + "and bu.updatedAt <= :cutoffDate";
+    parameters.put("errorCode", errorCode);
+    parameters.put("newStatus", newStatus);
+    parameters.put("updatedAt", now);
+    parameters.put("oldStatus", oldStatus);
+    parameters.put("cutoffDate", cutoffDate);
+
+    // Perform the update and get the count of updated records
+    int updatedCount = update(query, parameters);
```

Comment:
this method should return this long updatedCount only, not the list of entities.

Reply:
So we were discussing how do we log these updated rows. Won't it be helpful to get that information in log?

Reply:
I reverted the last commit, which brings it back to logging just the count.